packages/components/tool-bar/src/tool-bar.ts

```diff
@@ -0,0 +1,229 @@
+      'sl-menu-item-group': MenuItemGroup
+    };
+  }
+
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  // Observe changes to the size of the element.
+  #observer = new ResizeObserver(() => this.#onResize());
+
+  /** The horizontal alignment within the tool-bar. */
+  @property({ reflect: true }) align?: 'start' | 'end';
+
+  /** If true, the tool-bar is disabled and cannot be interacted with. */
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /** @internal True when the tool-bar is empty. */
+  @property({ type: Boolean, reflect: true }) empty?: boolean;
+
+  /** @internal The tool bar items. */
+  @state() items: ToolBarItem[] = [];
+
+  /** @internal The tool bar items that should be shown in the overflow menu. */
+  @state() menuItems: ToolBarItem[] = [];
+
+  /**
+   * If true, will cause the tool bar to not have a border; useful when embedding
+   * the tool-bar inside another component.
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'no-border' }) noBorder?: boolean;
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.setAttribute('role', 'toolbar');
+
+    this.#observer.observe(this);
+  }
+
+  override disconnectedCallback(): void {
+    this.#observer.disconnect();
+
+    super.disconnectedCallback();
+  }
+
+  override willUpdate(changes: PropertyValues<this>): void {
+    super.willUpdate(changes);
+
+    if (changes.has('disabled')) {
+      const children = this.renderRoot.querySelector('slot')?.assignedElements({ flatten: true }) ?? [];
+
+      if (this.disabled) {
+        children.forEach(el => el.setAttribute('disabled', ''));
+      } else {
+        children.forEach(el => el.removeAttribute('disabled'));
+      }
+    }
+
+    if (changes.has('items')) {
+      this.menuItems = this.items.filter(item => !item.visible);
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <div part="wrapper">
+        <slot @slotchange=${this.#onSlotChange}></slot>
+      </div>
+
+      ${this.menuItems.length
+        ? html`
+            <sl-menu-button>
+              <sl-icon name="far-ellipsis-vertical" slot="button"></sl-icon>
+              ${this.menuItems.map(item => this.renderMenuItem(item))}
+            </sl-menu-button>
+          `
+        : nothing}
+    `;
+  }
+
+  renderMenuItem(item: ToolBarItem): TemplateResult {
+    if (item.type === 'group') {
+      return html`
+        <sl-menu-item-group .heading=${item.label ?? ''} .selects=${item.selects}>
+          ${item.buttons.map(button => this.renderMenuItem(button))}
+        </sl-menu-item-group>
+        <hr />
+      `;
+    } else if (item.type === 'divider') {
+      return html`<hr />`;
+    } else {
+      return html`
+        <sl-menu-item @click=${() => item.click?.()} ?disabled=${item.disabled} ?selectable=${item.selectable}>
+          ${item.icon ? html`<sl-icon .name=${item.icon}></sl-icon>` : nothing} ${item.label}
```

Comment:
`aria-label` is missing here? :)

Reply:
How do you mean? If the menu item has text, it doesn't need `aria-label`, right?

Reply:
Is it necessary, since there is a label behind the icon?

Reply:
right, I didn't notice the label 😆